src/core/storage/sframe_data/groupby_aggregate_impl.cpp

```diff
@@ -217,128 +220,173 @@ size_t groupby_element::hash() const {
 /*                         group_aggregate_container                        */
 /*                                                                          */
 /****************************************************************************/
+
+/* static thread_local member initialization */
+thread_local group_aggregate_container::tls_segment_set
+    group_aggregate_container::tss_{};
+
 group_aggregate_container::group_aggregate_container(size_t max_buffer_size,
-                                                     size_t num_segments):
-    max_buffer_size(max_buffer_size), segments(num_segments) {
-  intermediate_buffer.open_for_write(num_segments);
-  for (size_t i = 0;i < segments.size(); ++i) {
-    segments[i].outiter = intermediate_buffer.get_output_iterator(i);
+                                                     size_t num_segments)
+    : max_buffer_size(max_buffer_size),
+      num_segments(num_segments),
+      task_id_(0),
+      gl_lock_pool_(num_segments),
+      gl_chunk_size_set_(num_segments) {
+  if (num_segments == 0) {
```

Comment:
This is an internal aspect, so an assert would be the right style here:  ASSERT_NE(num_segments, 0);

Reply:
Resolve this here.